framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -359,4 +361,38 @@ public boolean onReceivedClientCertRequest(CordovaWebView view, ICordovaClientCe
      */
     public void onConfigurationChanged(Configuration newConfig) {
     }
+
+    /**
+     * Called by the Plugin Manager when we need to actually request permissions
+     *
+     * @return              Returns the permission that was stored in the plugin
+     */
+
+    public String[] getPermissionRequest() {
+        return permissions;
+    }
+
+    public boolean hasPermisssion() {
+        for(String p : permissions)
+        {
+            if(PackageManager.PERMISSION_DENIED == cordova.getActivity().checkSelfPermission(p))
```

Comment:
+1 I do find this confusing. I don't also understand why geolocation plugin has to be different. I understand some of this has to be because of Chrome webview handling here: https://github.com/apache/cordova-android/pull/220/files#diff-eb00315f6b4fb38aba0833096a86bb8aR190

The plublic